src/main/java/com/gooddata/GoodData.java

```diff
@@ -167,21 +164,31 @@ public GoodData(String hostname, String login, String password, int port, GoodDa
      * @param settings additional settings
      */
     protected GoodData(String hostname, String login, String password, int port, String protocol, GoodDataSettings settings) {
-        notEmpty(hostname, "hostname");
-        notEmpty(login, "login");
-        notEmpty(password, "password");
-        notEmpty(protocol, "protocol");
-        httpClient = createHttpClient(login, password, hostname, port, protocol,
-                createHttpClientBuilder(settings));
+        this(
+                new GoodDataEndpoint(hostname, port, protocol),
+                new LoginPasswordAuthentication(login, password), settings
+        );
+    }
 
-        restTemplate = createRestTemplate(hostname, httpClient, port, protocol);
+    /**
+     * Create instance configured to communicate with GoodData Platform running on given endpoint and using
```

Comment:
sorry -it's not - the first letter is once capital and once not, even in this single javadoc.